src/server/stream_family.cc

```diff
@@ -603,6 +603,55 @@ int StreamTrim(const AddTrimOpts& opts, stream* s) {
   return 0;
 }
 
+template <typename T> struct PointerWrapper {
+  PointerWrapper(T* ptr) : ptr_{ptr} {
+  }
+
+  explicit operator bool() const {
+    return ptr_;
+  }
+
+  operator T*() {
+    return ptr_;
+  }
+
+  T* operator->() {
+    return ptr_;
+  }
+
+ protected:
+  T* ptr_;
+};
+
+struct GroupWrapper : public PointerWrapper<streamCG> {
+  explicit GroupWrapper(streamCG* cg) : PointerWrapper{cg} {
+  }
+
+  streamConsumer* FindConsumer(string_view name, sds& tmp_str, bool create = false) {
+    tmp_str = sdscpylen(tmp_str, name.data(), name.length());
+    auto* consumer = streamLookupConsumer(ptr_, tmp_str, SLC_NO_REFRESH);
+    if (consumer == nullptr && create)
+      return streamCreateConsumer(ptr_, tmp_str, NULL, 0, SCC_NO_NOTIFY | SCC_NO_DIRTIFY);
+    return consumer;
+  }
+};
+
+struct StreamWrapper : public PointerWrapper<stream> {
+  explicit StreamWrapper(stream* s) : PointerWrapper{s} {
+  }
+  explicit StreamWrapper(const CompactObj& co) : StreamWrapper{static_cast<stream*>(co.RObjPtr())} {
+  }
+
+  static void Create(CompactObj* obj) {
```

Comment:
I do not see how it's related to this class, can be just another helper function